### PR TITLE
NP-621: add doc for nodeport host addresses

### PR DESCRIPTION
**Pull request by zshi-redhat**
ovn-kubernetes doesn't restrict the host addresses where
k8s nodeport service can be accessed from outside microshift
node, this commit adds nftable instructions to drop packet
matching the nodeport and host interface IPs. These nftable
instructions can work regardless of firewalld service state.

Signed-off-by: Zenghui Shi <zshi@redhat.com>

### CODE REVIEW

File: docs/network/default_cni_plugin.md [hunk 1]
```diff
@@ -149,3 +150,44 @@ The specified interface will be added in OVS bridge `br-ex` which acts as gatewa
 microshift-ovs-init.service is able to setup one additional host interface for cluster ingress/egress traffic.
 This is done by specifying the `externalGatewayInterface` in the CNI config file `/etc/microshift/ovn.yaml`.
 The external gateway interface will be added in a second OVS bridge `br-ex1`. Cluster pod traffic destinated to additional host subnet will be routed through `br-ex1`.
+
+### NodePort host addresses
```
Comment (pmtk) [inline docs/network/default_cni_plugin.md:154]: Could we reword this?
I feel that something like "Blocking access to NodePort from/for specific host interfaces" would better describe intent of the section and the instructions here. What do you think?

File: docs/network/default_cni_plugin.md [hunk 2]
```diff
@@ -149,3 +150,44 @@ The specified interface will be added in OVS bridge `br-ex` which acts as gatewa
...
+> Record the `handle` number `56`, nodeport number `30700`
+
+<br>
+
+Insert a new rule before the ovn-kubernetes nodeport rule to `drop` the packet with matching destination port and ip:
+```text
+(host)$ nft -a insert rule ip nat OVN-KUBE-NODEPORT position 56 tcp dport 30700 ip daddr 192.168.150.33 drop
+```
+> Replace the handle number (after position) and nodeport number (after dport) with recorded values in the above step
+> Replace `192.168.150.33` with the IP address from the host interface where you'd like to block the nodeport service
```
Comment (pmtk) [inline docs/network/default_cni_plugin.md:175]:
````suggestion
```text
(host)$ HANDLE=56
(host)$ NODEPORT=30700
(host)$ INTERFACE_IP=192.168.150.33
(host)$ nft -a insert rule ip nat OVN-KUBE-NODEPORT position $HANDLE tcp dport $NODEPORT ip daddr $INTERFACE_IP drop
```
> Replace value of HANDLE and NODEPORT variables with recorded values in the above step
> Replace value of INTERFACE_IP with the IP address from the host interface where you'd like to block the NodePort service
````

WDYT?

File: docs/network/default_cni_plugin.md (hunk 1 above)
Comment (zshi-redhat) [reply docs/network/default_cni_plugin.md]: Done.

File: docs/network/default_cni_plugin.md (hunk 2 above)
Comment (zshi-redhat) [reply docs/network/default_cni_plugin.md]: sure, that looks clean and better.

File: docs/network/default_cni_plugin.md [hunk 3]
```diff
@@ -149,3 +150,47 @@ The specified interface will be added in OVS bridge `br-ex` which acts as gatewa
...
+```text
+(host)$ nft -a list table ip nat  | grep "chain OVN-KUBE-NODEPORT" -A3
+	chain OVN-KUBE-NODEPORT { # handle 22
+		tcp dport 30700 ip daddr 192.168.150.33 drop # handle 67
+		meta l4proto tcp fib daddr type local tcp dport 30700 counter packets 28 bytes 1680 dnat to 10.43.138.214:7000 # handle 56
+	}
+```
+
+> Record the `handle` number of the newly added rule (for removal)
+
```
Comment (pliurh) [inline docs/network/default_cni_plugin.md:192]: It would be better to also tell users how to make the nft rules permanent.

Comment (zshi-redhat) [reply docs/network/default_cni_plugin.md:186]: Done.
Updated the nftable rule to use PREROUTING chain which doesn't need to specify rule handle, this makes it easier to add a persist rule with nftables systemd service. Also added a link to the rhel9 doc about persisting and auto-loading nftable rules with systemd service.

File: docs/network/default_cni_plugin.md [hunk 4]
```diff
@@ -149,3 +150,43 @@ The specified interface will be added in OVS bridge `br-ex` which acts as gatewa
...
+<br>
+
+Insert a new rule in table `ip nat` chain `PREROUTING` to drop the packet with matching destination port and ip:
+```text
+(host)$ NODEPORT=30700
+(host)$ INTERFACE_IP=192.168.150.33
+(host)$ nft -a insert rule ip nat PREROUTING tcp dport $NODEPORT ip daddr $INTERFACE_IP drop
+```
+> Replace value of NODEPORT variable with the host port number assigned to kubernetes NodePort service
+> Replace value of INTERFACE_IP with the IP address from the host interface where you'd like to block the NodePort service
```
Comment (pmtk) [inline docs/network/default_cni_plugin.md:167]: I'm not sure about this, but I looked at "rich diff" and these two lines were merged into one. Maybe adding one empty with `>` will keep them separate but in the same quote block

```suggestion
> Replace value of NODEPORT variable with the host port number assigned to kubernetes NodePort service
>
> Replace value of INTERFACE_IP with the IP address from the host interface where you'd like to block the NodePort service
```

Comment (zshi-redhat) [reply docs/network/default_cni_plugin.md]: added line break at the end of each line.

**Pull request merged**